app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -134,10 +138,27 @@ class BrowserFragment : IWebViewLifecycleFragment(),
                 isReloadingForYoutubeDrawerClosed = true
                 reload()
             }
+            NavigationEvent.PIN_ACTION -> {
+                this@BrowserFragment.url?.let { url ->
+                    if (VAL_CHECKED == value) {
+                        CustomTilesAccessor.getInstance(context).pinSite(context, url)
+                        showCenteredToast(R.string.notification_pinned_site)
+                    } else {
+                        CustomTilesAccessor.getInstance(context).unpinSite(context, url)
+                        showCenteredToast(R.string.notification_unpinned_site)
+                    }
+                }
+            }
         }
         Unit
     }
 
+    private fun showCenteredToast(resId: Int) {
```

Comment:
nit: I think this should move to a helper to help keep BrowserFrag clean: `ToastUtils.showCentenedToast(resId: Int)`, for example.